docs/DeveloperGuide.md

```diff
@@ -4,26 +4,162 @@
+### Good Class
+#### Description
+The goods class keeps track of the various inventory that will be input into the system through the commands class.
+The diagram below shows the model component of the orderline class.
+![Good Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/Good.puml)
+
+The Goods class allows for the creation of orderline objects which has the following attributes: id, name, quantity and 
+description. Each attribute can be obtained using public get methods, and the attribute quantity can be set using the public set method.
+
+
+## Implementation
+### Regex Class
+#### Description
+The Regex Class is created to simplify the use of regular expressions in Java.
+![Regex Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/Regex.puml)
+The Class Diagram above shows the different variables and methods associated with the regex class.
+This class takes in the regular expression as a string and the string to be matched in the class contructor, and stores the results
+in the private variable `groupValues` which is a Hash Map where the key would be the capture group name and the value as the matched results.
+
+#### Operation
+![Regex Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/RegexMatching.puml)
+The above diagram shows the inner workings of how the Regex Class will work. Here are the steps taken by the Regex Class:
+1. findMatch() called when instantiating groupValues variable
+2. findGroups() called within findMatch() to find the capture group names within the regex string
+3. regexMatching() will then be called within findMatch() which will use Java's base Pattern and Matcher Class, and returns a Matcher Object
+4. findMatch() will then use the Matcher object to fill in the HashMap of <key, value> pair of <capture group name, regex matching>
+5. findMatch() will then return the HashMap object to groupValues variable
+
+#### Example of Usage
+```
+String regex = "id/(?<id>\\d*) n/(?<name>.*) q/(?<qty>\\d*) d/(?<desc>\\.*)";
+Regex regexMatch = new Regex(userInput, regex);
+HashMap<String, String> matches = regexMatch.getGroupValues();
+String id = matches.get("id");
+String name = matches.get("name");
+String quantity = matches.get("qty");
+String description = matches.get("desc");
+```
+### Commands Class
+#### Description
+The Commands Class is the class which contains all the available commands that can be used in the Simplst CLI application.
+![Commands Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/CommandsClass.puml)
+
+More about how each feature is run can be seen in each of the methods that can be found below.
```

Comment:
perhaps an explanation can be given for why this design of putting all commands into a single class was chosen rather than subclassing different commands.